source/extensions/quic_listeners/quiche/envoy_quic_proof_source.cc

```diff
@@ -0,0 +1,103 @@
+#include "extensions/quic_listeners/quiche/envoy_quic_proof_source.h"
+
+#include <openssl/bio.h>
+
+#include "envoy/ssl/tls_certificate_config.h"
+
+#include "extensions/quic_listeners/quiche/envoy_quic_utils.h"
+#include "extensions/quic_listeners/quiche/quic_io_handle_wrapper.h"
+#include "extensions/transport_sockets/well_known_names.h"
+
+#include "openssl/bytestring.h"
+#include "quiche/quic/core/crypto/certificate_view.h"
+
+namespace Envoy {
+namespace Quic {
+
+quic::QuicReferenceCountedPointer<quic::ProofSource::Chain>
+EnvoyQuicProofSource::GetCertChain(const quic::QuicSocketAddress& server_address,
+                                   const quic::QuicSocketAddress& client_address,
+                                   const std::string& hostname) {
+  absl::optional<std::reference_wrapper<const Envoy::Ssl::TlsCertificateConfig>> cert_config_ref =
+      getTlsCertConfig(server_address, client_address, hostname);
+  // Only return the first TLS cert config.
```

Comment:
Q: When will multiple certs exist? Why always return the first one?

Reply:
This comment should be removed.